### PR TITLE
feat: Add Submissions

### PR DESCRIPTION
**Pull request by gmeben**
This PR:

* Introduces the `Submission` model
* Establishes a one-to-many relationship between submissions and publications
* Establishes a many-to-many relationship between submissions and users
* Updates the GraphQL endpoint for submissions
    * submission - get a submission by ID
    * submissions - get all submissions
    * user.submissions - get all submissions for a user
    * publication.submissions - get all submissions for a publication
* Adds seeding for publications and submissions

### CODE REVIEW

File: backend/graphql/schema.graphql [hunk 1]
```diff
@@ -1,3 +1,12 @@
+#import permission.graphql
+#import role.graphql
+#import user.graphql
+#import social_media.graphql
+#import profile_metadata.graphql
+#import publication.graphql
+#import academic_profiles.graphql
+#import submission.graphql
```
Comment (Sruthin86) [inline backend/graphql/schema.graphql:8]:
```suggestion
#import academic_profiles.graphql
#import permission.graphql
#import profile_metadata.graphql
#import publication.graphql
#import social_media.graphql
#import submission.graphql
#import role.graphql
#import user.graphql
```

File: backend/graphql/submission.graphql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"""
+A Submission
+"""
+type Submission {
+    id: ID!
+    title: String
+    created_at: DateTime!
+    updated_at: DateTime
+    publication: publication
```
Comment (Sruthin86) [inline backend/graphql/submission.graphql:9]:
```suggestion
    publication: Publication
```

File: backend/tests/Feature/SubmissionTest.php [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            if ($random_role_id !== $submitter_id) {
+                $submission->users()->attach(
+                    $users->random(),
+                    [
+                        'role_id' => $submitter_id,
+                    ]
+                );
+            }
+        }
+        Submission::all()->map(function ($submission) use ($user_count) {
```
Comment (Sruthin86) [inline backend/tests/Feature/SubmissionTest.php:80]: - Should we add a test to check if the user for a submission is present in the users table?
- Should we add a test to check if the role for a submission is present in the roles table?

**Review by Sruthin86 — CHANGES_REQUESTED** [review]

File: backend/tests/Feature/SubmissionTest.php (hunk 3 above)
Comment (gmeben) [reply backend/tests/Feature/SubmissionTest.php]: Added the suggested assertions

**Review by wreality — CHANGES_REQUESTED** [review]

**Review by Sruthin86 — APPROVED** [review]
Looked at it during team time! Approving this pull request.

**Comment by Sruthin86** [comment]
Looked at it during team time! Approving this pull request.

**Pull request merged**